README.md
<!-- 
This README describes the package. If you publish this package to pub.dev,
this README's contents appear on the landing page for your package.

For information about how to write a good package README, see the guide for
[writing package pages](https://dart.dev/guides/libraries/writing-package-pages). 

For general information about developing packages, see the Dart guide for
[creating packages](https://dart.dev/guides/libraries/create-library-packages)
and the Flutter guide for
[developing packages and plugins](https://flutter.dev/developing-packages). 
-->

<<<<<<< HEAD
A new Flutter package that provide You A simple TimeLine

##Usage

[Example] (https://github.com/AbdllahMohamed/simple_timeline/blob/main/example/example.dart)

To use this package : *add the dependency to your ['pubspec.yaml'] file
=======
Simple Flutter TimeLine
>>>>>>> 684e59f469e332ade19c176bfb4bc6e26b07cf9e

## Features

https://i.ibb.co/VLV9Kny/Screenshot-108.png

## Getting started

 dependencies:
        flutter:
            sdk : flutter
        simple_timeline: ^0.0.1

## Usage

TODO: Include short and useful examples for package users. Add longer examples
to `/example` folder. 

```dart
import 'package:flutter/material.dart';

void main() async {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({
    Key? key,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Simpe_TimeLine',
      home:
          SimpleTimeLine(itemCount: 3, label: 'label', labelData: 'labelData',),
    );
  }
}

```

## Additional information

TODO: Tell users more about the package: where to find more information, how to 
contribute to the package, how to file issues, what response they can expect 
from the package authors, and more.


CHANGELOG.md
## 0.0.1

* A Simple Flutter TimeLine 


lib/src/simple_timeline.dart
import 'package:flutter/material.dart';

class SimpleTimeLine extends StatelessWidget {
  final int itemCount;
  Widget? nodeWidget;
  double? nodewidth;
  double? nodeHeight;
  Color? nodeColor;

  Widget? lineWidget;
  double? lineHeight;
  double? linewidth;
  Color? lineColor;
  MainAxisAlignment? mainAxisAlignment;
  CrossAxisAlignment? crossAxisAlignment;

  final String label;

  final String labelData;
  final TextStyle? labelTextStyle;
  final TextStyle? labelDataTextStyle;
  SimpleTimeLine({
    Key? key,
    required this.itemCount,
    required this.label,
    required this.labelData,
    this.labelTextStyle,
    this.labelDataTextStyle,
    this.nodeWidget,
    this.nodeHeight,
    this.nodewidth,
    this.nodeColor,
    this.lineWidget,
    this.lineHeight,
    this.linewidth,
    this.lineColor,
    this.mainAxisAlignment,
    this.crossAxisAlignment,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return ListView.builder(
      shrinkWrap: true,
      itemCount: itemCount,
      itemBuilder: (context, index) => Row(
        crossAxisAlignment: crossAxisAlignment ?? CrossAxisAlignment.start,
        mainAxisAlignment: mainAxisAlignment ?? MainAxisAlignment.start,
        children: [
          Padding(
            padding: const EdgeInsets.only(left: 8.0),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                nodeWidget ??
                    Container(
                      width: nodewidth ?? 50,
                      height: nodeHeight ?? 50,
                      decoration: BoxDecoration(
                        shape: BoxShape.circle,
                        color: nodeColor ?? Colors.black,
                      ),
                    ),
                index == itemCount - 1
                    ? const SizedBox()
                    : lineWidget ??
                        Container(
                          margin: const EdgeInsets.only(left: 22.0),
                          alignment: Alignment.center,
                          height: lineHeight ?? 60,
                          width: linewidth ?? 5,
                          decoration: BoxDecoration(
                            color: lineColor ?? Colors.black,
                          ),
                        ),
              ],
            ),
          ),
          Padding(
            padding: const EdgeInsets.only(left: 20.0),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(
                  label,
                  style: labelTextStyle ??
                      const TextStyle(
                        fontSize: 18,
                        color: Colors.black,
                      ),
                ),
                Padding(
                  padding: const EdgeInsets.only(left: 2.0),
                  child: Text(
                    labelData,
                    overflow: TextOverflow.clip,
                    style: labelDataTextStyle ??
                        const TextStyle(
                          fontSize: 18,
                          color: Colors.black,
                        ),
                  ),
                ),
              ],
            ),
          )
        ],
      ),
    );
  }
}


lib/src/custome_timeline.dart
import 'package:flutter/material.dart';

class CustomeTimeline extends StatelessWidget {
  final int itemCount;
  Widget? nodeWidget;
  double? nodewidth;
  double? nodeHeight;
  Color? nodeColor;

  Widget? lineWidget;
  double? lineHeight;
  double? linewidth;
  Color? lineColor;
  MainAxisAlignment? mainAxisAlignment;
  CrossAxisAlignment? crossAxisAlignment;

  final String label;

  final String labelData;
  final TextStyle? labelTextStyle;
  final TextStyle? labelDataTextStyle;
  CustomeTimeline({
    Key? key,
    required this.itemCount,
    required this.label,
    required this.labelData,
    this.labelTextStyle,
    this.labelDataTextStyle,
    this.nodeWidget,
    this.nodeHeight,
    this.nodewidth,
    this.nodeColor,
    this.lineWidget,
    this.lineHeight,
    this.linewidth,
    this.lineColor,
    this.mainAxisAlignment,
    this.crossAxisAlignment,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return ListView.builder(
      shrinkWrap: true,
      itemCount: itemCount,
      itemBuilder: (context, index) => Row(
        crossAxisAlignment: crossAxisAlignment ?? CrossAxisAlignment.start,
        mainAxisAlignment: mainAxisAlignment ?? MainAxisAlignment.start,
        children: [
          Padding(
            padding: const EdgeInsets.only(left: 8.0),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                nodeWidget ??
                    Container(
                      width: nodewidth ?? 50,
                      height: nodeHeight ?? 50,
                      decoration: BoxDecoration(
                        shape: BoxShape.circle,
                        color: nodeColor ?? Colors.black,
                      ),
                    ),
                index == itemCount - 1
                    ? const SizedBox()
                    : lineWidget ??
                        Container(
                          margin: const EdgeInsets.only(left: 22.0),
                          alignment: Alignment.center,
                          height: lineHeight ?? 60,
                          width: linewidth ?? 5,
                          decoration: BoxDecoration(
                            color: lineColor ?? Colors.black,
                          ),
                        ),
              ],
            ),
          ),
          Padding(
            padding: const EdgeInsets.only(left: 20.0),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(
                  label,
                  style: labelTextStyle ??
                      const TextStyle(
                        fontSize: 18,
                        color: Colors.black,
                      ),
                ),
                Padding(
                  padding: const EdgeInsets.only(left: 2.0),
                  child: Text(
                    labelData,
                    overflow: TextOverflow.clip,
                    style: labelDataTextStyle ??
                        const TextStyle(
                          fontSize: 18,
                          color: Colors.black,
                        ),
                  ),
                ),
              ],
            ),
          )
        ],
      ),
    );
  }
}


example/example.dart
import 'package:flutter/material.dart';

void main() async {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({
    Key? key,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Simpe_TimeLine',
      home:
          CustomeTimeline(itemCount: 3, label: 'label', labelData: 'labelData'),
    );
  }
}

class CustomeTimeline extends StatelessWidget {
  final int itemCount;
  Widget? nodeWidget;
  double? nodewidth;
  double? nodeHeight;
  Color? nodeColor;

  Widget? lineWidget;
  double? lineHeight;
  double? linewidth;
  Color? lineColor;
  MainAxisAlignment? mainAxisAlignment;
  CrossAxisAlignment? crossAxisAlignment;

  final String label;

  final String labelData;
  final TextStyle? labelTextStyle;
  final TextStyle? labelDataTextStyle;
  CustomeTimeline({
    Key? key,
    required this.itemCount,
    required this.label,
    required this.labelData,
    this.labelTextStyle,
    this.labelDataTextStyle,
    this.nodeWidget,
    this.nodeHeight,
    this.nodewidth,
    this.nodeColor,
    this.lineWidget,
    this.lineHeight,
    this.linewidth,
    this.lineColor,
    this.mainAxisAlignment,
    this.crossAxisAlignment,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: ListView.builder(
          shrinkWrap: true,
          itemCount: itemCount,
          itemBuilder: (context, index) => Row(
            crossAxisAlignment: crossAxisAlignment ?? CrossAxisAlignment.start,
            mainAxisAlignment: mainAxisAlignment ?? MainAxisAlignment.start,
            children: [
              Padding(
                padding: const EdgeInsets.only(left: 8.0),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    nodeWidget ??
                        Container(
                          width: nodewidth ?? 50,
                          height: nodeHeight ?? 50,
                          decoration: BoxDecoration(
                            shape: BoxShape.circle,
                            color: nodeColor ?? Colors.black,
                          ),
                        ),
                    index == itemCount - 1
                        ? const SizedBox()
                        : lineWidget ??
                            Container(
                              margin: const EdgeInsets.only(left: 22.0),
                              alignment: Alignment.center,
                              height: lineHeight ?? 60,
                              width: linewidth ?? 5,
                              decoration: BoxDecoration(
                                color: lineColor ?? Colors.black,
                              ),
                            ),
                  ],
                ),
              ),
              Padding(
                padding: const EdgeInsets.only(left: 20.0),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      label,
                      style: labelTextStyle ??
                          const TextStyle(
                            fontSize: 18,
                            color: Colors.black,
                          ),
                    ),
                    Padding(
                      padding: const EdgeInsets.only(left: 2.0),
                      child: Text(
                        labelData,
                        overflow: TextOverflow.clip,
                        style: labelDataTextStyle ??
                            const TextStyle(
                              fontSize: 18,
                              color: Colors.black,
                            ),
                      ),
                    ),
                  ],
                ),
              )
            ],
          ),
        ),
      ),
    );
  }
}
